src/commands/lightning/dev/site.ts

```diff
@@ -63,6 +72,46 @@ export default class LightningDevSite extends SfCommand<void> {
       }
 
       const selectedSite = new ExperienceSite(org, siteName);
+
+      if (!ssr) {
+        let sfdxProjectRootPath = '';
+        try {
+          sfdxProjectRootPath = await SfProject.resolveProjectPath();
+        } catch (error) {
+          throw new Error(sharedMessages.getMessage('error.no-project', [(error as Error)?.message ?? '']));
+        }
+        const previewUrl = await selectedSite.getPreviewUrl();
+        const username = connection.getUsername();
+        if (!username) {
+          throw new Error(sharedMessages.getMessage('error.username'));
+        }
+
+        this.log('Configuring local web server identity');
+        const appServerIdentity = await PreviewUtils.getOrCreateAppServerIdentity(connection);
+        const ldpServerToken = appServerIdentity.identityToken;
+        const ldpServerId = appServerIdentity.usernameToServerEntityIdMap[username];
+        if (!ldpServerId) {
+          throw new Error(sharedMessages.getMessage('error.identitydata.entityid'));
+        }
+
+        this.log('Determining the next available port for Local Dev Server');
+        const serverPorts = await PreviewUtils.getNextAvailablePorts();
+        this.log(`Next available ports are http=${serverPorts.httpPort} , https=${serverPorts.httpsPort}`);
+
+        this.log('Determining Local Dev Server url');
+        const ldpServerUrl = PreviewUtils.generateWebSocketUrlForLocalDevServer(Platform.desktop, serverPorts);
+        this.log(`Local Dev Server url is ${ldpServerUrl}`);
+
+        const logger = await Logger.child(this.ctor.name);
+        await startLWCServer(logger, sfdxProjectRootPath, ldpServerToken, Platform.desktop, serverPorts);
+        const url = new URL(previewUrl);
+        url.searchParams.set('aura.lwcDevServerUrl', ldpServerUrl);
+        url.searchParams.set('aura.lwcDevServerId', ldpServerId);
+        url.searchParams.set('lwc.mode', 'dev');
+        await open(url.toString());
+        return;
+      }
```

Comment:
nit: could we separate this out in into two different function calls? Run method then just calls previewHMR() / previewSSR() based on the cli arg being there?

Reply:
Done